### ISSUE TITLE
Wrong working directory for dependency with --terragrunt-source option

### ISSUE THREAD

**Issue by d-helios** [gruntwork-io/terragrunt#1124]
# Version
terragrunt version v0.22.5
# Use case
I use before_hook to download **override.tf** file from s3 to override some default behaviour of terraform module that I use. 

Example of my terragrunt.hcl

```
include {
  path = find_in_parent_folders()
}

terraform {
  source = "git::https://bitbucket.org/XXXXXXX/tf-modules.git//datalake?ref=${local.common_vars["tf_modules_version"]}"

  before_hook "before_hook_1" {
    commands = get_terraform_commands_that_need_vars()
    execute = [
      "aws",
      "s3",
      "cp",
      "s3://${local.artifact_bucket}/${local.git_repository}/releases/${local.artifacts_version}/${local.current_folder}.tf",
    "override.tf"]
    run_on_error = false
  }
}

locals {
  // map that hold common variables for all modules
  common_vars_file  = "${get_terragrunt_dir()}/${find_in_parent_folders("common_vars.yaml")}"
  common_vars       = yamldecode(file(local.common_vars_file))
  tfmodules_version = local.common_vars["cmdb"]["tf-modules"]

  // get git repository name for current dataproduct
  git_repository = basename(dirname(get_terragrunt_dir()))
  current_folder = basename(get_terragrunt_dir())

  // get the artifact version for current dataproduct
  artifacts_version          = local.common_vars["cmdb"][local.current_folder]

  // global artifact bucket
  artifact_bucket = local.common_vars["artifact_bucket"]
}

inputs = {
  artifacts_version          = local.artifacts_version
  local_tags = {
    Layer              = local.current_folder
    DataProduct        = local.git_repository
    DataProductVersion = local.artifacts_version
    TFModulesVersion   = local.tfmodules_version
  }
}
```

# What's wrong
*expected behaviour (it works if I do not specify the --terragrunt-source option)*
1. terragrunt create **.terragrunt-cache** directory
2. terragrunt copy modules specify in **source** keyword to the **.terragrunt-cache**
3. terragrunt change working directory to the **.terragrunt-cache/<hash_1>/<hash2>/submodule
4. terragrunt [before_hooks]
5. terragrunt [apply]
6. terragrunt [after_hooks]

*behaviour with --terragrunt-source option*
1. terragrunt create **.terragrunt-cache** directory
2. terragrunt copy modules specify in **source** keyword to the **.terragrunt-cache**
3. terragrunt change working directory to the **.terragrunt-cache/<hash_1>/<hash2>/submodule
4. terragrunt use the same submodule for all dependencies
5. execution failed

# Steps to reproduce
1. [Archive.zip](https://github.com/gruntwork-io/terragrunt/files/4446744/Archive.zip)
2. cd live/app_2
3. terragrunt apply-all --terragrunt-non-interactive  [Works fine]
4. terragrunt apply-all --terragrunt-non-interactive --terragrunt-source ~/tmp/path_to_example/modules [before_hook for app_1 executed in a wrong working directory ]

![2](https://user-images.githubusercontent.com/16616414/78713364-d6ee1e80-7922-11ea-935b-476f4d96fb00.png)
![1](https://user-images.githubusercontent.com/16616414/78713369-d8b7e200-7922-11ea-8402-8c9bf7fc3fe0.png)

**Comment by yorinasub17** [comment]
Thanks for providing a minimal reproducible case. I was able to identify the bug and work on a fix in #1134

**Comment by yorinasub17** [comment]
Fixed in https://github.com/gruntwork-io/terragrunt/releases/tag/v0.23.9. Please read the release notes for more info.
